### PR TITLE
feat(artblocks): Add support for artblocks mint plugin

### PR DESCRIPTION
**Pull request by mikesizz**
## Integration of `Art Blocks` mint action plugin for QuestDK
This PR introduces the integration of the `artblocks` mint plugin for QuestDK.

### About The Project
[Art Blocks](https://www.artblocks.io) is an NFT marketplace that allows users to mint generative artwork from a published "project". Projects are made up of artist provided logic that generates unique pieces of artwork given a set of customizable parameters. Each project allows for a fixed number of mints, and each mint operation is provided with a distinct set of parameters via contract-based random number generation.

### Implementation Details
This plugin aims to provide support for all shared minter suite contracts that are [compatible and active flagship V3 contracts](https://github.com/ArtBlocks/artblocks-contracts/blob/main/packages/contracts/MINTER_SUITE.md#active-flagship-minting-contracts). By targeting the latest V3 contracts we can ensure that all current implementations of Art Blocks are supported.

### Current Limitations
- Targeting minting contracts is on a per-minter basis to allow for targeting of specific purchase methodologies (direct purchase, dutch auction, etc).
- `tokenId` refers to the target Art Blocks `projectId` thus allowing targeting of minting on specific projects, not the output token ID.
- Only actively deployed Ethereum mainnet and Arbitrum mainnet V3 compatible contracts are supported.
  - Deployments: https://github.com/ArtBlocks/artblocks-contracts/tree/main/packages/contracts/deployments/minters

### CODE REVIEW

File: packages/registry/src/index.ts [hunk 1]
```diff
@@ -2,47 +2,48 @@ import {
   type ActionParams,
```
Comment (mikesizz) [inline packages/registry/src/index.ts:2]: Large change set due to running formatter. Adding artblocks plugin was the only meaningful change.

File: packages/artblocks/src/abi.ts [hunk 2]
```diff
@@ -0,0 +1,698 @@
+const PURCHASE_SET_PRICE_V5_ABI = [
```
Comment (mikesizz) [inline packages/artblocks/src/abi.ts:1]: Any contracts that do not have their own ABI are already covered by an existing ABI.

File: packages/artblocks/src/test-transactions.ts [hunk 3]
```diff
@@ -0,0 +1,338 @@
...
+    purchaseToSetPriceMerkleV5: PURCHASE_TO_SET_PRICE_MERKLE_V5,
+    purchaseSetPriceV4: PURCHASE_SET_PRICE_V4,
+    purchaseToSetPriceV4: PURCHASE_TO_SET_PRICE_V4,
+    purchaseSetPriceERC20V4: PURCHASE_SET_PRICE_ERC20_V4,
+    purchaseMerkleV5: PURCHASE_MERKLE_V5,
+    purchaseToMerkleV5: PURCHASE_TO_MERKLE_V5,
+    purchaseDAExpV4: PURCHASE_DA_EXP_V4,
+    purchaseToDAExpV4: PURCHASE_TO_DA_EXP_V4,
+  },
+  Arbitrum: {
```
Comment (mikesizz) [inline packages/artblocks/src/test-transactions.ts:270]: Arbitrum ABIs match new V3 contract ABIs. This is merely testing that transactions on Arbitrum are passing as expected. Deeper coverage would be handled by overall ABI coverage.

File: packages/artblocks/src/test-transactions.ts [hunk 4]
```diff
@@ -0,0 +1,338 @@
+import { type MintActionParams } from '@rabbitholegg/questdk'
+
+import {
+  type TestCase,
+  type TestParams,
+  createTestCase,
+} from '@rabbitholegg/questdk-plugin-utils'
+
+import { NULL_ADDRESS } from './const'
```
Comment (mmackz) [inline packages/artblocks/src/test-transactions.ts:9]: Just an fyi, viem has a built in constant called zeroAddress you can use for this purpose

**Review by Quazia — APPROVED** [review]
Now that you're on the team let's get this merged in! Was just waiting, don't like the idea of pulling in the code if things didn't pan out 😊

File: packages/artblocks/package.json [hunk 5]
```diff
@@ -0,0 +1,41 @@
+{
+  "name": "@rabbitholegg/questdk-plugin-artblocks",
+  "version": "1.0.0-alpha.11",
```
Comment (mmackz) [inline packages/artblocks/package.json:3]: Please reset the version back to 1.0.0-alpha.11 (This version number is for the template package)

File: packages/artblocks/CHANGELOG.md
Comment (mmackz) [inline packages/artblocks/CHANGELOG.md:1]: Wipe the changelog. It will be automatically generated once it is merged.

Just a blank file is fine.

**Review by mmackz — APPROVED** [review]
Requested a few small changes, but otherwise looks good to merge.

**Pull request merged**